src/middleware/services.ts

```diff
@@ -6,7 +6,7 @@ import { getFileService } from '../utils/get-file-service';
 
 // initialise any request-scoped services required by the app and store them on the request object for later use
 // see @types/express/index.d.ts for details
-export const initServices = (req: Request, res: Response, next: NextFunction): void => {
+export const initServices = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
```

Comment:
This looks like you might've tried adding the pool to the request object already and then reverted maybe?